modules/Library/library_manage_catalog_addProcess.php

```diff
@@ -28,23 +28,21 @@
     header("Location: {$URL}");
 } else {
     //Get general fields
-    $gibbonLibraryTypeID = $_POST['gibbonLibraryTypeID'];
-    $id = $_POST['idCheck'];
-    $name = $_POST['name'];
-    $producer = $_POST['producer'];
-    $vendor = $_POST['vendor'];
-    $purchaseDate = null;
-    if ($_POST['purchaseDate'] != '') {
-        $purchaseDate = dateConvert($guid, $_POST['purchaseDate']);
-    }
-    $invoiceNumber = $_POST['invoiceNumber'];
-    $imageType = $_POST['imageType'];
+    $gibbonLibraryTypeID = $_POST['gibbonLibraryTypeID'] ?? '';
+    $id = $_POST['idCheck'] ?? '';
+    $name = $_POST['name'] ?? '';
+    $producer = $_POST['producer'] ?? '';
+    $vendor = $_POST['vendor'] ?? '';
+    $purchaseDate = dateConvert($guid, $_POST['purchaseDate'] ?? null);
```

Comment:
`dateConvert` is deprecated since v16. It's probably best to rewrite `dateConvert` to `Gibbon\Services\Format::dateConvert` instead.